reproman/resource/session.py

```diff
@@ -707,11 +739,23 @@ def get_local_session(env={'LC_ALL': 'C'}, pty=False, shared=None):
         session.set_envvar(env)
     return session
 
+
 def get_updated_env(env, update):
     """Given an environment and set of updates, return updated one
 
     Special handling -- keys with None for the value, will be removed
     """
+    # If an update env var references itself, merge the original env value
+    # into the update value.
+    # (This is to somewhat generalize the "PATH=/foo:$PATH" special case.)
+    for k in update:
+        if k not in env or not isinstance(update[k], str) or \
+                not isinstance(env[k], str):
+            continue
+        token = "${}".format(k)
+        if token in update[k]:
+            update[k] = update[k].replace(token, env[k])
```

Comment:
Hmm, I think this approach is pretty brittle in terms of quoting and partial matches.  For example,

```
>       assert get_updated_env({'a': '/foo', 'b': 2}, {'a': '/bar:$apple'}) \
            == {'a': '/bar:$apple', 'b': 2}
E       AssertionError: assert {'a': '/bar:/foopple', 'b': 2} == {'a': '/bar:$apple', 'b': 2}
E         Omitting 1 identical items, use -vv to show
E         Differing items:
E         {'a': '/bar:/foopple'} != {'a': '/bar:$apple'}
```